src/averageAges.js

```diff
@@ -20,6 +20,23 @@ function calculateMenAverageAge(people, century) {
   // avoid using loop and forEach
   // replace `if ()` statement with &&, || or ?:
   // without nesting
+  const males = people.filter(person => person.sex === 'm');
+  const sumOfAge = males.reduce((totalAge, currentMale) => {
+    return totalAge + (currentMale.died - currentMale.born);
+  }, 0);
+
+  const malesInCentury = males.filter(person => {
+    return Math.ceil(person.died / 100) === century;
+  });
+
+  const sumOfAgeWithCentury = malesInCentury.reduce((totalAge, currentMale) => {
+    return totalAge + (currentMale.died - currentMale.born);
+  }, 0);
+
+  const avarageAge = sumOfAge / males.length;
+  const avrageAgeWithCentury = sumOfAgeWithCentury / malesInCentury.length;
+
+  return century ? avrageAgeWithCentury : avarageAge;
```

Comment:
maybe try different approach here.
First, depending on the `century` value determine the array of men you want to count the avg age for (use ternary operator already here), and then calculate the average age for this group that you got from that check. It will be much more concise